backend/graph/resolver.go

```diff
@@ -1,9 +1,61 @@
 package graph
 
+import (
+    "context"
+    "encoding/json"
+    "io/ioutil"
+    "strings"
+	"log"
+	"fmt"
+    "study-gator-backend/graph/model"
+)
+
 //go:generate go run github.com/99designs/gqlgen generate
 
 // This file will not be regenerated automatically.
 //
 // It serves as dependency injection for your app, add any dependencies you require here.
 
 type Resolver struct{}
+
+func (r *queryResolver) SearchByClassTitle(ctx context.Context, term string) ([]*model.Resource, error) {
+    titleLower := strings.ToLower(term)
+	
+	log.Printf("SearchByClassTitle query received: %s", term)
+
+	// Read from the JSON file 
+	bytes, err := ioutil.ReadFile("graph/resources.json") 
+	if err != nil {
+		log.Printf("Error reading resources.json file: %s", err)
+		return nil, fmt.Errorf("error reading from resources file: %w", err)
+	}
+
+	// Deserialize the JSON data into the slice of Resource
+	var resources []*model.Resource
```

Comment:
Shouldn't do this, by making these pointers you allow for null entries into that JSON, this will likely result in undefined behaviour if someone were to add null to that JSON.